cmp/grid/columns/ColumnGroup.js

```diff
@@ -16,8 +16,13 @@ import {startCase, isEmpty, castArray, clone} from 'lodash';
 export class ColumnGroup {
     /**
      * @param {Object} c - ColumnGroup configuration.
-     * @param {string} [c.groupId] - unique identifier for the ColumnGroup within its grid.
-     * @param {string} [c.headerName] - display text for grid header.
+     * @param {string} c.groupId - unique identifier for the ColumnGroup within its grid.
+     * @param {ColumnGroup~headerNameFn|string} [c.headerName] - display text for column group header.
```

Comment:
Sorry to be a sticker, but think that if we have a defined type in this file, all this text describing the function can be included in the docstring for that type.  The only description here should be 'display text' or 'display text for this header'

Also, could generalize the description of that type, and use a single 'Column~headerNameFn' as the type for Both ColumnGroup and Column

Reply:
There is a slight difference between the two, the Column `headerNameFn` is passed a `column` in the params, ColumnGroup `headerNameFn` is passed a `columnGroup`.

Do we care about that specificity in the param name? Or should we just always pass a `column` prop which could be a Column or ColumnGroup? Or have a single callback doc with both as optional? Couldn't decide when writing this initially, hence the separate callback defs.

Reply:
In this case, I think you actually get *more* info from using a single doc item for these two things.  It really is the same callback structure, and it actually gives the user more information to see that we think of it as the same object.

Don't know what the magic rule is, but in this case, seems like DRY and clarity both point toward a single doc.